src/ShoppingCenter.java

```diff
@@ -35,8 +39,76 @@ void updateShop(int shopId, Shop modifiedShop) {
     public List<Shop> findShopsByName(String name) {
         return shopList.stream()
                 .filter(shop -> shop.name.equals(name))
-                .collect(Collectors.toList())
+                .collect(Collectors.toList());
+    }
+
+    void printAllShops(){
+        for (int i = 0; i<this.shopList.size(); i++){
+            Shop current;
+            current = shopList.get(i);
+            System.out.println(current.name);
+        }
+    }
+
+
+    public List<Product> getAllProducts(){
+        List allProducts = new LinkedList();
+        for (int i = 0; i<this.shopList.size(); i++){
+            Shop current = shopList.get(i);
+            for (int j = 0; j<current.productsList.size(); j++){
+                ProductDetails currentProduct = current.productsList.get(j);
+                allProducts.add(currentProduct.product);
+            }
+        }
+        return allProducts;
+    }
+
+    public List<Service> getAllServices(){
+        List allServices = new LinkedList();
+        for (int i = 0; i<this.shopList.size(); i++){
+            Shop current = shopList.get(i);
+            for (int j = 0; j<current.servicesList.size(); j++){
+                Service currentService = current.servicesList.get(j);
+                allServices.add(currentService);
+            }
+        }
+        return allServices;
+    }
+
+    public List<String> getAllShopTypes(){
+        List allTypes = new LinkedList();
+        for (int i = 0; i<this.shopList.size(); i++){
+            Shop current = shopList.get(i);
+            for (int j = 0; j<current.types.size(); j++){
+                String currentType = current.types.get(j);
+                allTypes.add(currentType);
+            }
+        }
+        return allTypes;
     }
 
+    public Set<String> getAllShopTypes2() {
```

Comment:
....this looks great!

actually you may remove old `getAllShopTypes` and leave this one (remember to remove "2" from name). But if you want to keep them both, no problem (then only replace for with foreach loops :P )